app/components/primer/beta/flash.rb

```diff
@@ -6,65 +6,108 @@ module Beta
       #
-      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
-      renders_one :action, lambda { |**system_arguments|
-        deny_tag_argument(**system_arguments)
-        system_arguments[:tag] = :div
-        system_arguments[:classes] = class_names(system_arguments[:classes], "flash-action")
+      # To render a button, call the `with_action_button` method, which accepts the arguments accepted by <%= link_to_component(Primer::Beta::Button) %>.
+      #
+      # To render custom content, call the `with_action_content` method and pass a block that returns HTML.
+      renders_one :action, types: {
+        button: lambda { |**system_arguments|
+          deny_tag_argument(**system_arguments)
+
+          Primer::Beta::Button.new(**system_arguments)
+        },
 
-        Primer::BaseComponent.new(**system_arguments)
+        content: lambda { |**system_arguments|
+          deny_tag_argument(**system_arguments)
+          system_arguments[:tag] = :div
+
+          Primer::BaseComponent.new(**system_arguments)
+        }
       }
 
       DEFAULT_SCHEME = :default
       SCHEME_MAPPINGS = {
         DEFAULT_SCHEME => "",
-        :warning => "flash-warn",
-        :danger => "flash-error",
-        :success => "flash-success"
+        :warning => "Banner--warning",
+        :danger => "Banner--error",
+        :success => "Banner--success"
+      }.freeze
+
+      DEFAULT_ICONS = {
+        default: :bell,
+        warning: :alert,
+        danger: :stop,
+        success: :"check-circle"
       }.freeze
+
       # @example Schemes
-      #   <%= render(Primer::Beta::Flash.new) { "This is a flash message!" } %>
-      #   <%= render(Primer::Beta::Flash.new(scheme: :warning)) { "This is a warning flash message!" } %>
-      #   <%= render(Primer::Beta::Flash.new(scheme: :danger)) { "This is a danger flash message!" } %>
-      #   <%= render(Primer::Beta::Flash.new(scheme: :success)) { "This is a success flash message!" } %>
+      #   <div style="display: grid; row-gap: 15px">
+      #     <%= render(Primer::Beta::Flash.new) { "This is a flash message!" } %>
+      #     <%= render(Primer::Beta::Flash.new(scheme: :warning)) { "This is a warning flash message!" } %>
+      #     <%= render(Primer::Beta::Flash.new(scheme: :danger)) { "This is a danger flash message!" } %>
+      #     <%= render(Primer::Beta::Flash.new(scheme: :success)) { "This is a success flash message!" } %>
+      #   </div>
       #
       # @example Full width
       #   <%= render(Primer::Beta::Flash.new(full: true)) { "This is a full width flash message!" } %>
       #
       # @example Dismissible
-      #   <%= render(Primer::Beta::Flash.new(dismissible: true)) { "This is a dismissible flash message!" } %>
+      #   <%= render(Primer::Beta::Flash.new(dismissible: true, reappear: true)) { "This is a dismissible flash message!" } %>
       #
-      # @example Icon
+      # @example Custom icon
```

Comment:
Can we change this to be the default icons per variant?

Reply:
@langermank if you don't provide an icon, it defaults to one based on the scheme. In other words, if you don't specify which icon you want and the scheme is `:danger`, it will automatically choose the [`:stop` Octicon](https://primer.style/octicons/stop-24). Is that what you had in mind?